### DISCUSSION TITLE
Sklearn-linear model_locator for a Cross Site Model Evaluation

### DISCUSSION THREAD

**Discussion by kandalto** [NVIDIA/NVFlare#2455] [Q&A]
### Python version (`python3 -V`)

3.10.13

### NVFlare version (`python3 -m pip list | grep "nvflare"`)

2.4

### NVFlare branch (if running examples, please use the branch that corresponds to the NVFlare version, `git branch`)

2.4

### Operating system

Ubuntu 20.04

### Have you successfully run any of the following examples?

- [ ] hello-numpy-sag with simulator
- [ ] hello-pt with simulator
- [ ] hello-numpy-sag with POC
- [ ] hello-pt with POC

### Please describe your question

Hello!

I want to add Cross Site Model Evaluation (CSE) (CrossSiteModelEval worflow) to the *sklearn-linear* example here: 
***https://github.com/NVIDIA/NVFlare/blob/main/examples/advanced/sklearn-linear/sklearn_linear_higgs.ipynb***

However, I cannot find a readily available *model_locator/formatter* for *sklearn-linear* in the codebase. Going through the other examples of CSE I found CSE *model_locators* for pytorch and numpy models at 
***https://github.com/NVIDIA/NVFlare/blob/main/nvflare/app_opt/pt/file_model_locator.py*** and  ***https://github.com/NVIDIA/NVFlare/blob/main/nvflare/app_common/np/np_model_locator.py***

I suspect these files can be modified for *sklearn-linear* models and put in the custom folder of the *config_fed_server* file. However, it is not intuitive for me which changes I should make. Can you please confirm that these are the files than needs to be modified? What modifications should I make? Do I need just *model_locator* or *formatter* as well?

Maybe there are other already implemented solutions that I am missing?

Many thanks.

**Comment by chesterxgchen** [comment]
Thanks for your question, sorry for the delayed response. @ZiyueXu77  can you help answer this question

**Reply by ZiyueXu77** [reply]
Hi @kandalto, thanks for the question, and yes I think you can start with the numpy one, which is more general-purpose
https://github.com/NVIDIA/NVFlare/blob/main/nvflare/app_common/np/np_model_locator.py
You will need to modify it so as to satisfy sklearn's model format for linear model, please refer to these lines
https://github.com/NVIDIA/NVFlare/blob/dev/examples/advanced/sklearn-linear/jobs/sklearn_linear_base/app/custom/linear_learner.py#L67-L70
